### PR TITLE
Create oop library

### PR DESCRIPTION
**Pull request by devMunyi**
- Created a class **Person** with the following:

1. Instance vars: @id, @name, and @age.
2. Constructor with name, age, and parent_permission as parameters. name and parent_permission are optional and have default values of "Unknown" and true.
3. Getters for @id, @name, and @age.
4. Setters for @name and @age.
5. Private method of_age? that returns true if @age is greater or equal to 18 and false otherwise.- 
6. Public method can_use_services? that returns true if a person is of age or if they have permission from parents.

- Created class **Student** with the following:

1. Inherits from Person.
2. Constructor extends parent's constructor by adding @classroom and a parameter for it.
3. Method play_hooky that returns "¯\(ツ)/¯".

- Created class **Teacher** with the following:

1. Inherits from Person.
2. Constructor extends parent's constructor by adding @specialization and a parameter for it.
3. Override can_use_services? so it always returns true.

- Each class should be saved in a separate file namely person.rb, student.rb, and teacher.rb

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require './person'
+
+class Student < Person
+  # contructor method
+  def initialize(classroom)
+    super(27, 'Harry Winks')
+    @classroom = classroom
+  end
```
Comment (Nemwel-Boniface) [inline student.rb]: - [ ] The constructor extends the parent's constructor by adding @classroom and a parameter for it. Also, when creating the super method, you are supposed to extend the functionalities from the Parent class. Kindly consider doing it like this:

```
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require './person'
+
+class Teacher < Person
+  # contructor method
+  def initialize(specialization)
+    super(35, 'John Doe')
+    @specialization = specialization
+  end
```
Comment (Nemwel-Boniface) [inline teacher.rb]: - [ ] The constructor extends the parent's constructor by adding @classroom and a parameter for it. Also, when creating the super method, you are supposed to extend the functionalities from the Parent class.  Kindly implement it just as I have suggested to you above.

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @devMunyi            :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
Hi @devMunyi , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Optional suggestions


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**